en/08_Changelogs/6.0.0.md

```diff
@@ -29,6 +30,242 @@ title: 6.0.0 (unreleased)
+-        my-action: 'Perform my custom action in dev/my-action (do not run in CLI)'
+-    my-other-action:
+-      controller: 'App\Dev\MyOtherActionController'
+-      links:
+-        my-other-action: 'Perform my custom action in dev/my-other-action'
++  controllers:
++    my-action:
++      class: 'App\Dev\MyActionController'
++      description: 'Perform my custom action in dev/my-action'
++  commands:
++    my-other-action: 'App\Dev\MyOtherActionCommand'
+```
+
+```diff
+ namespace App\Dev;
+
+-use SilverStripe\Control\Controller;
+-use SilverStripe\Control\Director;
+-use SilverStripe\Control\HTTPRequest;
++use SilverStripe\Dev\Command\DevCommand;
+-use SilverStripe\Dev\DevelopmentAdmin;
++use SilverStripe\PolyExecution\PolyOutput;
+-use SilverStripe\Security\Permission;
+ use SilverStripe\Security\PermissionProvider;
+-use SilverStripe\Security\Security;
++use Symfony\Component\Console\Command\Command;
++use Symfony\Component\Console\Input\InputInterface;
++use Symfony\Component\Console\Input\InputOption;
+
+-class MyOtherActionController extends Controller implements PermissionProvider
++class MyOtherActionController extends DevCommand implements PermissionProvider
+ {
++    protected static string $commandName = 'app:my-other-action';
++
++    protected static string $description = 'Perform my custom action in dev/my-other-action or via sake app:my-other-action';
++
++    private static array $permissions_for_browser_execution = [
++        'MY_CUSTOM_PERMISSION',
++    ];
++
++    public function getTitle(): string
++    {
++        return 'My other action';
++    }
++
+-    protected function init(): void
+-    {
+-        parent::init();
+-
+-        if (!$this->canInit()) {
+-            Security::permissionFailure($this);
+-        }
+-    }
+-
+-    public function index(HTTPRequest $request)
++    protected function execute(InputInterface $input, PolyOutput $output): int
+     {
+-        $someVar = $request->getVar('some-var');
++        $input->getOption('some-var');
+-        if (Director::is_cli()) {
+-            $body = "some output\n";
+-        } else {
+-            $body = "some output<br>\n";
+-        }
++        $output->writeln('some output');
+-
+-        return $this->getResponse()->setBody($body);
++        return Command::SUCCESS;
+     }
+
++    public function getOptions(): array
++    {
++        return [
++            new InputOption('some-var', null, InputOption::VALUE_NONE, 'some get variable'),
++        ];
++    }
+-    public function canInit(): bool
+-    {
+-        return (
+-            Director::isDev()
+-            || (Director::is_cli() && DevelopmentAdmin::config()->get('allow_all_cli'))
+-            || Permission::check('MY_CUSTOM_PERMISSION')
+-        );
+-    }
+
+     // ...
+ }
+```
+
+You would now access the `/dev/my-action` action via an HTTP request only. The `/dev/my-other-action` action can be access via an HTTP request, or by using `sake app:my-other-action` on the command line.
```

Comment:
```suggestion
You would now access the `/dev/my-http-only-action` action via an HTTP request only. The `/dev/my-http-and-cli-action` action can be access via an HTTP request, or by using `sake app:my-http-and-cli-action` on the command line.
```

Reply:
Done.... I think this is a duplicate suggestion?